qiskit_optimization/problems/variable.py

```diff
@@ -148,3 +148,17 @@ def as_tuple(self) -> Tuple[str, Union[float, int], Union[float, int], VarType]:
             variable type.
         """
         return self.name, self.lowerbound, self.upperbound, self.vartype
+
+    def __repr__(self):
+        if self._vartype == VarType.BINARY:
+            return f"<Variable: {self._name} {self._vartype.name.lower()}>"
+        lowerbound = "" if self._lowerbound == -INFINITY else f"{self._lowerbound} <= "
+        upperbound = "" if self._upperbound == INFINITY else f" <= {self._upperbound}"
+        return f"<Variable: {lowerbound}{self._name}{upperbound} {self._vartype.name.lower()}>"
```

Comment:
Since repr is just str plus the Variable text you not do repr much easier as something like
```
return f"<Variable: {str(self)}>"
```
to save replicating the logic.

Reply:
Thanks. I will do it.